holoviews/plotting/bokeh/util.py

```diff
@@ -146,18 +148,64 @@ def layout_padding(plots):
         expanded_plots.append([])
         for c, p in enumerate(row):
             if p is None:
-                p = Figure(plot_width=widths[c],
-                           plot_height=heights[r])
-                p.text(x=0, y=0, text=[' '])
+                x_range = Range1d(start=0, end=1)
+                y_range = Range1d(start=0, end=1)
+                p = Figure(plot_width=widths[c], plot_height=heights[r],
+                           x_range=x_range, y_range=y_range)
                 p.xaxis.visible = False
                 p.yaxis.visible = False
-                p.outline_line_color = None
-                p.xgrid.grid_line_color = None
-                p.ygrid.grid_line_color = None
+                p.outline_line_alpha = 0
+                p.grid.grid_line_alpha = 0
             expanded_plots[r].append(p)
     return expanded_plots
 
 
+def make_axis(axis, size, factors, dim, flip=False, rotation=0):
+    factors = list(map(dim.pprint_value, factors))
+    nchars = np.max([len(f) for f in factors])
+    ranges = FactorRange(factors=factors)
+    ranges2 = Range1d(start=0, end=1)
+    axis_label = dim_axis_label(dim)
+
+    rotation = np.radians(rotation)
+    if axis == 'x':
+        align = 'center'
+        # Adjust height to compensate for label rotation
+        height = int(50 + np.abs(np.sin(rotation)) * (nchars*8))
```

Comment:
Sure, might be good to parameterize it actually, if you have really long axis labels the layout solver could blow up I suppose and this would let you have some control.